src/standalone/Snow/neural_snow/ModelTools.jl

```diff
@@ -0,0 +1,323 @@
+                l3 = Dense(nfeatures, 1),
+            ),
+            vcat, #returns [predicted value, input...]
+        ),
+        get_boundaries = Parallel(
+            vcat,
+            up_bound = x -> relu.(x[1, :])' .* (x[p_idx + 1, :] .> 0)', # = upper = relu(upper)
+            low_bound = x -> x[z_idx + 1, :]', # = z = relu(z)
+            output_pos = x -> x[1, :]',
+        ),
+        final_scale = Dense(
+            Matrix{dtype}(diagm([1.0, 1.0, 1.0])),
+            false,
+            identity,
+        ),
+        #output_no_thresh = x -> x[3, :]'
+        apply_relus = Dense(get_relus, false, relu),  #returns relu(upper) = upper, relu(z) = z, relu(upper - pred_z)
+        apply_upper = Dense(get_min, false, relu), #returns relu(min(pred_z, upper)+z), relu(z) = z
+        apply_lower = Dense(get_max, false, identity), #returns relu(min(pred_z, upper)+z) - relu(z) = max(min(pred_Z, upper), lower)
+    )
+    return model
+end
+# Do we want to modify this to take a list of input features, and the names of the depth and precipitation variables,
+# instead of specifying nfeatures, z_idx, p_idx? Same functionality but simplifies the usage pipeline.
+
+
+"""
+    get_model_ps(model)
+
+Return the trainable weights for the developed neural model.
+
+# Arguments
+- `model::Chain`: the neural model to be used.
+"""
+function get_model_ps(model::Chain)
+    return Flux.params(Flux.params(model[:pred])[2:7])
+end
+
+"""
+    settimescale!(model, dt; dtype)
+
+Set the timescale parameter for model usage.
+
+# Arguments
+- `model::Chain`: the neural model to be used.
+- `dt::Real`: the number of seconds per timestep for usage.
+- `dtype::Type`: Sets type, consistent with neural model. Default is Float32.
+"""
+function settimescale!(model, dt::Real; dtype::Type = Float32)
+    model[:final_scale].weight[2, 2] = dtype(1.0 / dt)
+end
+
+"""
+    setoutscale!(model, scale; dtype)
+
+Set the physical scaling parameter for model usage (i.e. rectifying scaling done on model input).
+
+# Arguments
+- `model::Chain`: the neural model to be used.
+- `scale::Real`: the scaling parameter to return data to applicable units.
+- `dtype::Type`: Sets type, consistent with neural model. Default is Float32.
+"""
+function setoutscale!(model, scale::Real; dtype::Type = Float32)
+    model[:final_scale].weight[3, 3] = dtype(scale)
+end
+
+"""
+    LRmodel(data, vars, target; dtype, scale_const)
+
+Create a linear regression model on a data frame for comparison to neural model.
+Returns the coefficients for the model.
+
+# Arguments
+- `data::DataFrame`: The data set to be utilized.
+- `vars::Vector{Symbol}`: The input variables to be used in the model creation.
+-  `target::Symbol`: The target variable to be used in the model ceation.
+- `dtype::Type`: Sets type, consistent with neural model. Default is Float32.
+- `scale_const`: Optional scaling constant for model output. Default is 1.0.
+"""
+function LRmodel(
```

Comment:
if we do need a special type for a linear model, perhaps we can call it a `LinearModel` instead for clarity?